### ISSUE TITLE
h2spec 5.1-11 requires GOAWAY frame

### ISSUE THREAD

**Issue by bryancall** [summerwind/h2spec#88]
When testing with ATS 8.0.0/master and h2spec 2.1.0 it is failing 5.1-11 and requires a GOAWAY frame.  However, the HTTP/2 RFC states that the server should send a stream error when there is a data frame on a close stream:

https://tools.ietf.org/html/rfc7540#section-6.1:
> If a DATA frame is received
> whose stream is not in "open" or "half-closed (local)" state, the
> recipient MUST respond with a stream error (Section 5.4.2) of type
> STREAM_CLOSED.

My test:

```
16:27:23 homer:/usr/local/var/log/trafficserver$ h2spec  -t -k -p 4443 http2/5.1/11 -v
Hypertext Transfer Protocol Version 2 (HTTP/2)
  5. Streams and Multiplexing
    5.1. Stream States
           [send] SETTINGS Frame (length:6, flags:0x00, stream_id:0)
           [recv] SETTINGS Frame (length:12, flags:0x00, stream_id:0)
           [send] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
           [recv] WINDOW_UPDATE Frame (length:4, flags:0x00, stream_id:0)
           [recv] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
           [send] HEADERS Frame (length:15, flags:0x05, stream_id:1)
           [recv] HEADERS Frame (length:109, flags:0x04, stream_id:1)
           [recv] DATA Frame (length:531, flags:0x01, stream_id:1)
           [send] DATA Frame (length:4, flags:0x01, stream_id:1)
           [recv] RST_STREAM Frame (length:4, flags:0x00, stream_id:1)
           [recv] Timeout
      × 11: closed: Sends a DATA frame
        -> The endpoint MUST treat this as a connection error of type STREAM_CLOSED.
           Expected: GOAWAY Frame (Error Code: STREAM_CLOSED)
                     Connection closed
             Actual: RST_STREAM Frame (length:4, flags:0x00, stream_id:1)

Failures:

Hypertext Transfer Protocol Version 2 (HTTP/2)
  5. Streams and Multiplexing
    5.1. Stream States
      × 11: closed: Sends a DATA frame
        -> The endpoint MUST treat this as a connection error of type STREAM_CLOSED.
           Expected: GOAWAY Frame (Error Code: STREAM_CLOSED)
                     Connection closed
             Actual: RST_STREAM Frame (length:4, flags:0x00, stream_id:1)

Finished in 2.0229 seconds
1 tests, 0 passed, 0 skipped, 1 failed
```

**Comment by summerwind** [comment]
Thank you for the report. 

This is a *"double definition"* in RFC. 
I'm going to fix to wait a stream error on http2/5.1/11.

https://tools.ietf.org/html/rfc7540#section-5.1:

> Similarly, an endpoint that receives any frames after receiving a frame with the END_STREAM flag set MUST treat that as a connection error (Section 5.4.1) of type STREAM_CLOSED

https://tools.ietf.org/html/rfc7540#section-6.1:

> If a DATA frame is received whose stream is not in "open" or "half-closed (local)" state, the recipient MUST respond with a stream error (Section 5.4.2) of type  STREAM_CLOSED.

**Comment by bryancall** [comment]
Version 2.1.1 is passing all tests now:

```
10:03:44 homer:(8.0.x)~/dev/apache/trafficserver$ h2spec  -t -k -p 4443 http2/5.1/11 -v
Hypertext Transfer Protocol Version 2 (HTTP/2)
  5. Streams and Multiplexing
    5.1. Stream States
           [send] SETTINGS Frame (length:6, flags:0x00, stream_id:0)
           [recv] SETTINGS Frame (length:12, flags:0x00, stream_id:0)
           [send] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
           [recv] WINDOW_UPDATE Frame (length:4, flags:0x00, stream_id:0)
           [recv] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
           [send] HEADERS Frame (length:15, flags:0x05, stream_id:1)
           [recv] HEADERS Frame (length:109, flags:0x04, stream_id:1)
           [recv] DATA Frame (length:531, flags:0x01, stream_id:1)
           [send] DATA Frame (length:4, flags:0x01, stream_id:1)
           [recv] RST_STREAM Frame (length:4, flags:0x00, stream_id:1)
      ✔ 11: closed: Sends a DATA frame

Finished in 0.0237 seconds
1 tests, 1 passed, 0 skipped, 0 failed
```
